### PR TITLE
Fixes and enhancement 

### PR DESCRIPTION
**Pull request by wolfy2k**
filmParser.js
- fixed id obtaining
- splited title and type (possible breaking change)
- introduced type
- introduced poster
- fixed length (it returns an episode length in case of tv show)

searchParser.js
- introduced year, searchName and type
- fixed null result when csfd returns the movie directly after search (eg. https://www.csfd.cz/hledat/?q=Amerikáni)

### CODE REVIEW

**Comment by PatrikValkovic** [comment]
Pull request was noticed and will be merge ASAP.

**Comment by wolfy2k** [comment]
I should probably fix the tests, but I did not find a while for that :-D
Also since the split of title and type, we might consider it as major version, so we dont break someones code...

**Comment by PatrikValkovic** [comment]
Types were striped from the title (at least in tests). Did you find entity, for which it will return different results?

**Comment by PatrikValkovic** [comment]
Can you please refer to the situation, when `if (timeWhole.indexOf('x') != -1 )` condition in filmParser.parseLength is true? I was unable to find film with "x" inside of the string.

**Comment by wolfy2k** [comment]
That is a tv show.

File: lib/parsers/searchParser.js [hunk 1]
```diff
@@ -6,30 +6,53 @@
...
-    const idInString = href.substring(href.indexOf('/', 1) + 1, href.indexOf('-'))
-    const id = parseInt(idInString)
-    const name = $(el).text().trim()
-    return {id, name}
-  })
-  let moreFilms = []
-  if (moreFilmsContainer !== null)
-    moreFilms = $(moreFilmsContainer).find('li a').map((i, el) => {
-      const href = $(el).attr('href')
+  const profileFilmContainer = $("#profile");
```
Comment (PatrikValkovic) [inline lib/parsers/searchParser.js]: What is purpose of this statement with following condition?

**Comment by wolfy2k** [comment]
fixed null result when csfd returns the movie directly after search (eg. https://www.csfd.cz/hledat/?q=Amerikáni)

**Pull request merged**